src/main/scala/viper/gobra/translator/implementations/translator/ProgramsImpl.scala

```diff
@@ -45,15 +47,17 @@ class ProgramsImpl extends Programs {
           case p => Violation.violation(s"found unsupported member: $p")
         }
       }
+      memberWriter.map(m => (m, ctx))
     }
 
     val progW = for {
-      memberss <- sequence(program.members map goM)
-      members = memberss.flatten
+      membersWithCtxs <- sequence(program.members map goM)
+      ctxs = membersWithCtxs map { case (_, ctx) => ctx }
+      members = membersWithCtxs flatMap { case (m, _) => m }
```

Comment:
You could use unzip and then flatten.

Reply:
Thanks, that's exactly what I was looking for. Somehow I did not spot `unzip`